### DISCUSSION TITLE
Build failing for correct field type in OAuth Provider response

### DISCUSSION THREAD

**Discussion by onlinejudge95** [nextauthjs/next-auth#11302] [Help] [answered]
I am using Google for Oauth and have implemented the `signIn` callback to upsert the user to my DB post successful signIn.

Here is my implementation for the same
```ts
async signIn({profile}) {
    if (!profile?.email) {
        throw new Error('No email associated with the email profile');
    }

    await prisma.users.upsert({
        where: {
            email: profile.email
        },
        create: {
            id: profile.sub as string,
            email: profile.email,
            name: profile.name,
            avatar: profile.picture,
        },
        update: {
            name: profile.name,
            avatar: profile.picture,
        },
    });
    return true;
}
```
I receive an error when doing a build
```
./src/app/api/auth/[...nextauth]/route.ts:33:27
Type error: Property 'picture' does not exist on type 'Profile'.

  31 |           email: profile.email,
  32 |           name: profile.name,
> 33 |           avatar: profile.picture,
     |                           ^
  34 |         },
  35 |         update: {
  36 |           name: profile.name,
error Command failed with exit code 1.
```
But if I replace the `picture` field with `image` it resolves the build error but I don't get the field updated in my DB, any clue what I am doing wrong, here is my prisma definition of the `User` model
```prisma
model Users {
  id        String   @id @unique
  email     String   @unique
  name      String?
  avatar    String?
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt

  @@map("users")
}
```
Here are the redacted logs for debug purpose https://gist.github.com/onlinejudge95/5bc5f50e78d548661de47621e37d5dd8

### EDIT
Adding debug logs to prove that picture field does exist in the response schema

**Comment by k-taro56** [comment]
Why did you decide to use `profile.picture` instead of `profile.image` in the first place?

**Reply by onlinejudge95** [reply]
> But if I replace the picture field with image it resolves the build error but I don't get the field updated in my DB

Please read the question correctly

**Reply by k-taro56** [reply]
Sorry, I couldn't read it, can you tell me where it is clearly stated?

**Reply by onlinejudge95** [reply]
Are you even reading my reply?

**Reply by k-taro56** [reply]
What I want to know is what made you decide to use a non-existent property name.
It doesn't matter if the resulting value is in the DB or not.

**Reply by onlinejudge95** [reply]
@k-taro56 I think you don't understand English well, please use a good translation service before you feel the urge to reply.

Firstly if you check my logs the field does exist in the `OAuthProfile` key, also if I run my code locally with `yarn dev` I can get the flow working correctly when using the `picture` field, and my OAuth workflow passes, my cookies are set and my DB upsert works fine.

On using the `image` field the same thing described above happens except that the DB upsert does not fill data in the `avatar` column.

When running the code using `yarn build` I get a failure (if I use the `picture` field) which is expected, my point then is why is my flow working fine when using the dev server.

**Reply by k-taro56** [reply]
I got it! Thank you.

**Comment by k-taro56** [comment]
This seems to be a problem with DB, not next-auth.

**Reply by onlinejudge95** [reply]
Please keep the replies in a single thread, also can you help in pointing that its not a next-auth issue?

**Reply by k-taro56** [reply]
I apologize for being rude as a beginner.
If there is just no data in the DB, isn't it obvious that that is not a problem with next-auth?

**Reply by onlinejudge95** [reply]
@k-taro56 db field gets populated if I use the `picture` field and does not get populated if I use the `image` field, but it gives the build error on the former.

**Reply by k-taro56** [reply, marked as answer]
I think this will work perfectly!
```ts
async signIn({ user }) {
    if (!user?.email) {
        throw new Error('No email associated with the email profile');
    }

    await prisma.users.upsert({
        where: {
            email: user.email
        },
        create: {
            id: user.id,
            email: user.email,
            name: user.name,
            avatar: user.image,
        },
        update: {
            name: user.name,
            avatar: user.image,
        },
    });
    return true;
}
```

**Reply by onlinejudge95** [reply]
@k-taro56 thanks for the help, this works.
